init.lua

```diff
@@ -786,6 +793,16 @@ function Form:show_hud(player, ctx)
     hud_fs.show_hud(player, self, tree)
 end
 
+local open_inv_formspecs = {}
+function Form:set_as_inventory_for(player, ctx)
+    local name = player:get_player_name()
+    -- Formname of "" is inventory
+    local fs, form_info = prepare_form(self, player, "", ctx or {})
```

Comment:
Maybe this function should preserve `ctx` if the same form is already set as the inventory?

```suggestion
    local old_form_info = open_inv_formspecs[name]
    if not ctx and old_form_info and old_form_info.self == self then
        ctx = old_form_info.ctx
    end

    -- Formname of "" is inventory
    local fs, form_info = prepare_form(self, player, "", ctx or {},
        old_form_info and old_form_info.auto_name_id)
```

`auto_name_id` stores the next number to use for automatic button naming to work around race conditions that would happen if button names were reused.

I may move the `auto_name_id < 1e6` check from `update_form` into `_render` later so it doesn't have to be duplicated here.